### PR TITLE
feat: Add "allow sharing" checkbox to video editor

### PR DESCRIPTION
**Pull request by jansenk**
Add a checkbox to enable the public video / video sharing flag on a video.

Screenshot 2023-03-09 at 1 03 02 PM
Screenshot 2023-03-09 at 1 03 08 PM

### CODE REVIEW

File: src/editors/containers/VideoEditor/components/VideoSettingsModal/components/hooks.test.js [hunk 1]
```diff
@@ -35,6 +35,7 @@ jest.mock('../../../../../data/redux', () => ({
       videoSource: (state) => ({ videoSource: state }),
       fallbackVideos: (state) => ({ fallbackVideos: state }),
       allowVideoDownloads: (state) => ({ allowVideoDownloads: state }),
+      allowVideoSharing: (state) => ({ allowVideoSharing: state }),
```
Comment (leangseu-edx) [inline src/editors/containers/VideoEditor/components/VideoSettingsModal/components/hooks.test.js:38]: add `state.testGetter(state.keys.allowVideoSharing);` to the test

**Review by leangseu-edx — APPROVED** [review]
One change requested, the rest look good.

**Review by kenclary — APPROVED** [review]
looks alright

**Comment by jansenk** [comment]
@kenclary It looks like I can't merge, only people with write privileges can merge. Would you mind squash+merging this for me?

**Pull request merged**